src/app-layer.c

```diff
@@ -66,6 +67,16 @@ struct AppLayerThreadCtx_ {
 #endif
 };
 
+typedef struct AppLayerCounters_ {
+    char *name;
```

Comment:
We construct the name later and we use it in StatsRegisterCounter which is not currently used a const so this would require some patching in these.